### PR TITLE
feat: PowerMushroomAndFlower

### PR DESCRIPTION
**Pull request by GastonGutierrezC**
## Description

Make the collections with the powers, this includes only the power of the mushroom and the fire flower for Mario.

## Changes Made

- [x] Create a specific class for Mario's transformation to his different states according to his powers.
- [x] An Enum class  to identify Mario's condition
- [x] Add nimation to handle the big Mario Sprites and mario with fire power.

## Checklist

- [x] Mario when colliding with the mushroom transforms into big Mario.
- [x] Mario when colliding with the fire flower transforms into Mario fire.
- [x] includes the crouching animation of Mario grande or Mario fire.

Code reviuw
@OropezaHugo 
@JuanPabloArequipa

### CODE REVIEW

**Review by OropezaHugo — DISMISSED** [review]

**Comment by JuanPabloArequipa** [comment]
Cuando Mario es pequeño y come el poder de la flor, no cambia su estado correctamente:


https://github.com/Nintentando-Team-Software-Developmet/MarioBrosGame/assets/135129192/759b8a3b-9a36-4565-9bdd-2eb0bd38e292

File: MarioGame/Source/Systems/BlockSystem.cs [hunk 1]
```diff
@@ -311,4 +312,13 @@ private static void ApplyRepulsionForce(Contact contact)
         bodyA.ApplyLinearImpulse(-repulsionForce * normal);
         bodyB.ApplyLinearImpulse(repulsionForce * normal);
     }
+
+    private static StatusMario getStatusMario(PlayerComponent playerComponentMario)
+    {
+
+        showStatusMario = playerComponentMario.statusMario;
+
+        return showStatusMario;
```
Comment (JuanPabloArequipa) [inline MarioGame/Source/Systems/BlockSystem.cs:321]: No es necesario crear un metodo para acceder a playerComponentMario.statusMario

File: MarioGame/Source/Systems/BlockSystem.cs [hunk 2]
```diff
@@ -290,6 +289,8 @@ private static bool IsCollisionAtBase(Contact contact, float blockBottomY, IEnum
         foreach (var playerEntity in entities)
         {
             var playerCollider = playerEntity.GetComponent<ColliderComponent>().collider;
+            var playerComponent = playerEntity.GetComponent<PlayerComponent>();
+            getStatusMario(playerComponent);
```
Comment (JuanPabloArequipa) [inline MarioGame/Source/Systems/BlockSystem.cs:293]: Tampoco utilizas el valor que retorna el metodo, me parece que esta demas

File: MarioGame/Source/Systems/MarioPowersSystem.cs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    private readonly Collection<Action> pendingActions = new Collection<Action>();
+    private static double invulnerabilityEndTime { get; set; }
+    private bool isInvulnerable { get; set; }
+
+    public override void Update(GameTime gameTime, IEnumerable<Entity> entities)
+    {
+        var playerEntities = entities.WithComponents(typeof(PlayerComponent), typeof(ColliderComponent), typeof(AnimationComponent));
+        var mushroomEntities = entities.WithComponents(typeof(MushroomComponent), typeof(ColliderComponent));
+        var flowerEntities = entities.WithComponents(typeof(FlowerComponent), typeof(ColliderComponent));
+        var enemyEntities = entities.WithComponents(typeof(EnemyComponent), typeof(ColliderComponent), typeof(AnimationComponent));
```
Comment (JuanPabloArequipa) [inline MarioGame/Source/Systems/MarioPowersSystem.cs:33]: Cada entidad tiene un componente especifico que lo distingue: PlayerComponent, MushroomComponent, etc no es necesario que busques los componentes comunes (ColliderComponent, AnimationComponent)

File: MarioGame/Source/Systems/BlockSystem.cs (hunk 1 above)
Comment (GastonGutierrezC) [reply MarioGame/Source/Systems/BlockSystem.cs]: Done

File: MarioGame/Source/Systems/BlockSystem.cs (hunk 2 above)
Comment (GastonGutierrezC) [reply MarioGame/Source/Systems/BlockSystem.cs]: Done

File: MarioGame/Source/Systems/MarioPowersSystem.cs (hunk 3 above)
Comment (GastonGutierrezC) [reply MarioGame/Source/Systems/MarioPowersSystem.cs]: Done

**Review by JuanPabloArequipa — APPROVED** [review]

**Review by OropezaHugo — APPROVED** [review]

**Pull request merged**